packages/fairmq/package.py

```diff
@@ -9,74 +12,76 @@ class Fairmq(CMakePackage):
+    version('1.4.11', tag='v1.4.11', commit='e6dede492e02d6966cfa68e6820f90f679b74592', submodules=True, no_cache=True)
+    version('1.4.10', tag='v1.4.10', commit='f31be6d7a10ab0b7e4183576438268c8122117a5', submodules=True, no_cache=True)
+    version('1.4.9', tag='v1.4.9', commit='f6e3183f453d3daf97e529d3b834da715b43c47b', submodules=True, no_cache=True)
+    version('1.4.8', tag='v1.4.8', commit='3f5374820a56c472a79e81b977861eddf4bff343', no_cache=True)
+    version('1.4.7', tag='v1.4.7', commit='7cacf471b957aa1a89616b84339ce0b25699c7a0', no_cache=True)
+    version('1.4.6', tag='v1.4.6', commit='93eb599df7d9ee4c869e2fe70c03b7a845739683', no_cache=True)
+    version('1.4.5', tag='v1.4.5', commit='b1c82641237c1da7ae4d7af1fcb1cf69f596a54c', no_cache=True)
+    version('1.4.4', tag='v1.4.4', commit='99c8d33191a8f69bcd6b51cfa8c11a1d736734b0', no_cache=True)
+    version('1.4.3', tag='v1.4.3', commit='3582091b1c76cbe689b8e66813ea71dd5ff55bc5', no_cache=True)
+    version('1.4.2', tag='v1.4.2', commit='2457094b6c4004a75620a45329574d983cdc263e', no_cache=True)
+    version('1.4.1', tag='v1.4.1', commit='95ec56dcf0d95917a0c9cdecd1004d4a1c4aa479', no_cache=True)
+    version('1.4.0', tag='v1.4.0', commit='4c2785dfc1d98576917bc3cfb2a01eab675c5f6e', no_cache=True)
+    version('1.3.9', tag='v1.3.9', commit='d91a7d23616d2781938423b2362651185e2c0607', no_cache=True)
+    # TODO Once https://github.com/spack/spack/issues/14344 is resolved, enable
+    #      source caching again (by removing the `no_cache` argument).
+
+    variant('build_type', default='RelWithDebInfo',
+            values=('Debug', 'Release', 'RelWithDebInfo'),
+            multi=False,
+            description='CMake build type')
+    variant('cxxstd', default='default',
+            values=('11', '14', '17'),
+            multi=False,
+            description='Force the specified C++ standard when building.')
+    conflicts('cxxstd=11', when='@1.4.11:')
+
+    patch('fix_find_dds.patch', when='@1.4.0:1.4.4')
+
+    depends_on('googletest@1.7:', when='@:1.4.8')
+    depends_on('boost@1.64: +container+program_options+thread+system+filesystem+regex+date_time', when='@1.3')
+    depends_on('boost@1.64: +container+program_options+filesystem+date_time+regex', when='@1.4,develop')
+    depends_on('fairlogger@1.2:1.5', when='@:1.4.7')
+    depends_on('fairlogger@1.2:', when='@1.4.8:,develop')
+    depends_on('zeromq@4.1.5:')
+    depends_on('nanomsg@1.1.5:')
+    depends_on('msgpack-c@3.1:')
+    depends_on('dds@2.2:2.4', when='@:1.4.9')
+    depends_on('dds@2.5-odc', when='@1.4.10')
+    depends_on('dds@3.0:', when='@1.4.11:,develop')
+    depends_on('flatbuffers', when='@1.4.9:,develop')
+    depends_on('pmix@2.1.4:', when='@1.4:,develop')
+
+    depends_on('cmake@3.9.4:', type='build', when='@1.3')
+    depends_on('cmake@3.10:', type='build', when='@1.4.0:1.4.7')
+    depends_on('cmake@3.11:', type='build', when='@1.4.8:')
+    depends_on('cmake@3.12:', type='build', when='@develop')
```

Comment:
Please remember to add a new line here for the next fairmq release, that depends on cmake 3.12.
Or could we just assume, that the next release is `1.4.13`? Then `when='@1.4.13:'` would be better?

Reply:
This is definitely something the package maintainer has to get right. While it is likely the next release is indeed `v1.4.13`, IMHO the package should not try to guess anything.